### PR TITLE
Unit Label display settings

### PR DESCRIPTION
**Pull request by SJuliez**
Adds some more options to unit label display.

Note that the reduced vehicle chassis names ("Ontos") are created by taking the chassis and removing from the end words from a list of generic words like Heavy, Tank, Defense until something non-generic is found. This works reasonably well for many units and there's no pressure for it to find really short names for all units. The first word of the name is never removed.

The label styles are cycled using the keybind and output a description to the chatbox so players know what the style shows even if they don't have a nicknamed unit at the moment. I intentionally didn't implement some client settings for it as there's too many of those already. Cycling through it is easy.

Other suggestions or combinations of features as below, please tell me.

The current style (still available): 
image

Abbreviated style: 
image

Chassis only:
image

Nickname and chassis:
image

Only nicknames:
image

and no labels (as before)

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/GUIPreferences.java [hunk 1]
```diff
@@ -347,6 +349,7 @@ protected GUIPreferences() {
         store.setDefault(GAME_SUMMARY_MINI_MAP, false);
         store.setDefault(ENTITY_OWNER_LABEL_COLOR, true);
         store.setDefault(UNIT_LABEL_BORDER, true);
+        store.setDefault(UNIT_LABEL_STYLE, "NICKNAME");
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/GUIPreferences.java:352]: Can we please use the enum .name method instead? Helps prevent issues with name changes (I've made this mistake in MekHQ before)

File: megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java [hunk 2]
```diff
@@ -71,7 +74,14 @@
...
     // Keep track of ECM state, as it's too expensive to compute on the fly.
     private boolean isAffectedByECM = false;
-
+    
+    private static final Set<String> removableNameStrings = new HashSet<String>( 
+            Arrays.asList("Defense", "Platoon", "Heavy", "Medium", "Light", "Artillery", "Tank", 
+                    "Wheeled", "Command", "Standard", "Hover", "Hovercraft", "Mechanized", 
+                    "(Standard)", "Platoon", "Defense", "Transport", "Vehicle", "Air", 
+                    "Assault", "Mobile", "Platform", "Battle Armor", "Vessel", "Infantry",
+                    "Fighting", "Fire", "Suport", "Reconnaissance", "Fast"));
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java:83]: Why not just keep it as an array? Also, new HashSet<String> is completely superfluous in modern Java, as you only need to specify the class (i.e. String) if it is a wildcard on the backend

File: megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java [hunk 3]
```diff
@@ -88,19 +98,96 @@ public EntitySprite(BoardView1 boardView1, final Entity entity,
...
+                if (!pilotNick().isBlank()) {
+                    return "\"" + pilotNick().toUpperCase() + "\"";
+                } else if (!unitNick().isBlank()) {
+                    return "\'" + unitNick() + "\'";
+                } else {
+                    return "";
+                }
+            default: // ONLY_STATUS
+                return "";
             }
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java:132]: Please indent the switch. It's a minor thing, but I've found that it actually really helps with reading.

File: megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java [hunk 4]
```diff
@@ -88,19 +98,96 @@ public EntitySprite(BoardView1 boardView1, final Entity entity,
...
-                        .toUpperCase();
+            switch (GUIPreferences.getInstance().getUnitLabelStyle()) {
+            case FULL:
+                return standardLabelName();
+            case ABBREV:
+                if (entity instanceof Mech) {
+                    return entity.getModel();
+                } else {
+                    return abbreviateUnitName(standardLabelName());
+                }
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java:111]: Ternary candidate

File: megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java [hunk 5]
```diff
@@ -88,19 +98,96 @@ public EntitySprite(BoardView1 boardView1, final Entity entity,
...
         }
     }
+    
+    /** 
+     * Returns a shortened unit name string, mostly for vehicles. Words contained
+     * in the removableNameStrings list are taken away from the end of the name
+     * until something is encountered that is not contained in that list. 
+     * On Mech names this will typically have no effect.
+     */
+    private static String reduceVehName(String unitName) {
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java:142]: Can we name this reduceVehicleName? Helps with future-proofing

**Review by Windchild292 — COMMENTED** [review]
Few things first

File: megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java (hunk 2 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java:83]: I'll change the initialization to Set.of(), much simpler. The reduceVehicleName() must find if a term is contained in that word list, which is simplest and fastest in a HashSet (although that millisecond won't be important)

**Comment by sixlettervariables** [comment]
@SJuliez did not expect that, [but you're correct](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html#of(E...))! The `IllegalArgumentException` thrown in the static initializer causes that ClassDefNotFound exception. What awful behavior...sets should just deduplicate:
> **Throws:**
> `IllegalArgumentException` - if there are any duplicate elements

File: megamek/src/megamek/MegaMek.java [hunk 6]
```diff
@@ -94,7 +94,7 @@ public static void main(String[] args) {
                 }
             }
 
-            configureLogging(logFileName);
+//            configureLogging(logFileName);
```
Comment (NickAragua) [inline megamek/src/megamek/MegaMek.java:97]: Is this commented out code something that needs to be retained?

**Review by NickAragua — COMMENTED** [review]
One question, looks fine otherwise.

File: megamek/src/megamek/MegaMek.java (hunk 6 above)
Comment (SJuliez) [reply megamek/src/megamek/MegaMek.java]: Ouch! Good you saw that. I always do that to get errors in the console. Corrected it.

**Pull request merged**